packages/product/src/services/helpers/retrieve-entity.ts

```diff
@@ -0,0 +1,47 @@
+import { FindConfig, DAL, Context } from "@medusajs/types"
+import { ModulesSdkUtils, MedusaError, isDefined } from "@medusajs/utils"
+import { lowerCaseFirst } from "@medusajs/utils"
+
+type RetrieveEntityParams<TDTO> = {
+  id: string,
+  entityName: string,
+  repository: DAL.TreeRepositoryService
+  config: FindConfig<TDTO>
+  sharedContext?: Context
+}
+
+export async function retrieveEntity<
```

Comment:
praise: Nice, one could argue this should live in utils?

Reply:
agree, in utils/src/modules-sdk?